src/Adapters/NLogTarget.Shared/ApplicationInsightsTarget.cs

```diff
@@ -1,202 +1,212 @@
+            }
+            else
+            {
+                this.SendTrace(logEvent);
+            }
+        }
+
+        private static string GetAssemblyVersion()
+        {
+            return typeof(ApplicationInsightsTarget).Assembly.GetCustomAttributes(false)
+                    .OfType<AssemblyFileVersionAttribute>()
+                    .First()
+                    .Version;
+        }
+
+        private void SendException(LogEventInfo logEvent)
+        {
+            var exceptionTelemetry = new ExceptionTelemetry(logEvent.Exception)
+            {
+                SeverityLevel = this.GetSeverityLevel(logEvent.Level)
+            };
+
+            this.BuildPropertyBag(logEvent, exceptionTelemetry);
+            this.telemetryClient.Track(exceptionTelemetry);
+        }
+
+        private void SendTrace(LogEventInfo logEvent)
+        {
+            string logMessage = this.Layout.Render(logEvent);
+            
+            var trace = new TraceTelemetry(logMessage)
+            {
+                SeverityLevel = this.GetSeverityLevel(logEvent.Level)
+            };
+
+            this.BuildPropertyBag(logEvent, trace);
+            this.telemetryClient.Track(trace);
+        }
+
+        private void BuildPropertyBag(LogEventInfo logEvent, ITelemetry trace)
+        {
+            trace.Timestamp = logEvent.TimeStamp;
+            trace.Sequence = logEvent.SequenceID.ToString(CultureInfo.InvariantCulture);
+
+            IDictionary<string, string> propertyBag;
+
+            if (trace is ExceptionTelemetry)
+            {
+                propertyBag = ((ExceptionTelemetry)trace).Properties;
+            }
+            else
+            {
+                propertyBag = ((TraceTelemetry)trace).Properties;
+            }
+
+            if (!string.IsNullOrEmpty(logEvent.LoggerName))
+            {
+                propertyBag.Add("LoggerName", logEvent.LoggerName);
+            }
+
+            if (logEvent.UserStackFrame != null)
+            {
+                propertyBag.Add("UserStackFrame", logEvent.UserStackFrame.ToString());
+                propertyBag.Add("UserStackFrameNumber", logEvent.UserStackFrameNumber.ToString(CultureInfo.InvariantCulture));
+            }
+
+            var properties = logEvent.Properties;
+            if (properties != null)
+            {
+                foreach (var keyValuePair in properties)
+                {
+                    string key = keyValuePair.Key.ToString();
+                    object valueObj = keyValuePair.Value;
+                    if (valueObj == null)
+                    {
+                        continue;
+                    }
+
+                    string value = valueObj.ToString();
+                    if (propertyBag.ContainsKey(key))
+                    {
+                        if (value == propertyBag[key])
+                        {
+                            continue;
+                        }
+                        key += "_1";

```

Comment:
Thanks for your comments.

I don't believe it is possible to have more than one duplicate here.  Please correct me if I'm wrong.  The reason this issue happens at all is because the code is specifically calling out a few property names (LoggerName, UserStackFrame and UserStackFrameNumber) to populate before copying from IDictionary (logEvent.Properties) to IDictionary (trace.Properties).  Or are you suggesting that this code should be prepared to handle an implementation of IDictionary that allows for duplicate keys?

RE: the diff issue...I honestly cannot figure out what went wrong there.  Pulling the original and the one I committed and comparing them w/ a hex editor for a few minutes shows that the tabs/spaces and line endings are the same.  Will spend more time on this issue shortly.